### PR TITLE
Add time partitioning field to google_bigquery_table resource

### PR DESCRIPTION
**Pull request by amatellanes**
Add time partitioning field to [google_bigquery_table](https://www.terraform.io/docs/providers/google/r/bigquery_table.html) resource. This field is available on type [`TimePartitioning`](https://godoc.org/cloud.google.com/go/bigquery#TimePartitioning) from package [`bigquery`](https://godoc.org/cloud.google.com/go/bigquery):

```go
type TimePartitioning struct {
    ...

    // If empty, the table is partitioned by pseudo column '_PARTITIONTIME'; if set, the
    // table is partitioned by this field. The field must be a top-level TIMESTAMP or
    // DATE field. Its mode must be NULLABLE or REQUIRED.
    Field string
}
```

### CODE REVIEW

File: google/resource_bigquery_table.go [hunk 1]
```diff
@@ -428,6 +440,11 @@ func expandTimePartitioning(configured interface{}) *bigquery.TimePartitioning {
 
 func flattenTimePartitioning(tp *bigquery.TimePartitioning) []map[string]interface{} {
 	result := map[string]interface{}{"type": tp.Type}
+	result["field"] = tp.Type
```
Comment (danawillow) [inline google/resource_bigquery_table.go]: shouldn't this be tp.Field?

**Comment by amatellanes** [comment]
@danawillow I have fix `flattenTimePartitioning` function to retrieve `field`. Also I have created `TestAccBigQueryTable_TimePartitioningField` test to check time partitioning field but I'm not really sure if I should have added this test to an existing one... perhaps `TestAccBigQueryTable_Basic`...

**Comment by danawillow** [comment]
Thanks! Yeah I think having it in `basic` would work fine. Do you mind also adding an import step to the test? We're trying to standardize on as many tests as possible having that step since it lets us check that all the fields match. It would look like this:
```
			{
				ResourceName:        "google_bigquery_table.test",
				ImportState:         true,
				ImportStateVerify:   true,
			},
```

**Comment by amatellanes** [comment]
Hi @danawillow 

Changes done.

**Comment by danawillow** [comment]
Hey @amatellanes, I'm not sure if you were intending to have the test update the existing table, but by having the two steps in the same test that's what it tried to do. It sounds like from this error message that you can't update that field. If that's true, can you make it `ForceNew`?

```
$ make testacc TEST=./google TESTARGS='-run=TestAccBigQueryTable_Basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google -v -run=TestAccBigQueryTable_Basic -timeout 120m
=== RUN   TestAccBigQueryTable_Basic
=== PAUSE TestAccBigQueryTable_Basic
=== CONT  TestAccBigQueryTable_Basic
--- FAIL: TestAccBigQueryTable_Basic (6.64s)
	testing.go:513: Step 2 error: Error applying: 1 error(s) occurred:

		* google_bigquery_table.test: 1 error(s) occurred:

		* google_bigquery_table.test: googleapi: Error 400: Cannot change partitioning spec for a partitioned table., invalid
FAIL
FAIL	github.com/terraform-providers/terraform-provider-google/google	6.847s
make: *** [testacc] Error 1
```

**Comment by amatellanes** [comment]
I've just marked the field as `ForceNew`. Since we cannot update partitioning field for a partitioned table, should I create a new test or modify `testAccBigQueryTable` to add the field `Field` to `time_partitioning` block?

**Comment by danawillow** [comment]
I would just modify `testAccBigQueryTable` but I don't care particularly strongly

**Comment by amatellanes** [comment]
Finally I've chosen to modify `testAccBigQueryTable`.

**Pull request merged**